blaze-server/src/main/scala/org/http4s/server/blaze/WebSocketDecoder.scala

```diff
@@ -42,5 +43,6 @@ private class WebSocketDecoder
     * @return optional message if enough data was available
     */
   @throws[TranscodeError]
+  @throws[ProtocolException]
   def bufferToMessage(in: ByteBuffer): Option[WebSocketFrame] = Option(bufferToFrame(in))
```

Comment:
I'm interested, why is this wrapped in an Option?

Reply:
It can null, which is Evil.  The `Option` is introduced in the wrong place, I think.

Reply:
Does your comment mandate any changes?

Reply:
No, I think it's the right thing for the issue at hand.  But that function that can return `null` should be refactored to not as separate work.